benchmarks/entropy_adiabat/plugins/entropy_advection.cc

```diff
@@ -0,0 +1,238 @@
+      const unsigned int n_q_points = scratch.finite_element_values.n_quadrature_points;
+      std::vector<double> residuals(n_q_points,0.0);
+
+      if (advection_field.is_temperature() ||
+          this->introspection().name_for_compositional_index(advection_field.compositional_variable) != "entropy")
+        return residuals;
+
+      this->get_heating_model_manager().evaluate(scratch.material_model_inputs,
+                                                 scratch.material_model_outputs,
+                                                 scratch.heating_model_outputs);
+
+      for (unsigned int q=0; q < n_q_points; ++q)
+        {
+          const Tensor<1,dim> u = (scratch.old_velocity_values[q] +
+                                   scratch.old_old_velocity_values[q]) / 2;
+
+          const double dField_dt = (this->get_old_timestep() == 0.0) ? 0.0 :
+                                   (
+                                     ((scratch.old_field_values)[q] - (scratch.old_old_field_values)[q])
+                                     / this->get_old_timestep());
+          const double u_grad_field = u * (scratch.old_field_grads[q] +
+                                           scratch.old_old_field_grads[q]) / 2;
+
+          const double density       = scratch.material_model_outputs.densities[q];
+          const double entropy_conductivity  = scratch.material_model_outputs.thermal_conductivities[q] *
+                                               scratch.material_model_inputs.temperature[q] /
+                                               scratch.material_model_outputs.specific_heat[q];
+          const double k_Delta_field = entropy_conductivity * (scratch.old_field_laplacians[q] +
+                                                               scratch.old_old_field_laplacians[q]) / 2;
+
+          const double gamma           = scratch.heating_model_outputs.heating_source_terms[q];
+
+          residuals[q]
+            = std::abs((density * scratch.material_model_inputs.temperature[q]) * (dField_dt + u_grad_field) - k_Delta_field - gamma);
+        }
+      return residuals;
+    }
+  }
+
+// The function below replaces the normal temperature advection assembler with the entropy advection assembler of this file
+  template <int dim>
+  void set_assemblers_entropy_advection(const SimulatorAccess<dim> &simulator_access,
+                                        Assemblers::Manager<dim> &assemblers)
+  {
+    AssertThrow (Plugins::plugin_type_matches<MaterialModel::EntropyModel<dim>>
+                 (simulator_access.get_material_model()),
+                 ExcMessage ("The entropy advection assembler can only be used with the "
+                             "material model 'entropy model'!"));
+
+    AssertThrow (assemblers.advection_system.size() == 1,
+                 ExcMessage("There is more than one advection assembler active. This is not allowed."));
```

Comment:
Currently this benchmark only allows a single advection field, which has to be entropy. If someone wants to advect other fields they need to check if that works (it would also ask the question what entropy they are on, and what the material model should do). So in essence: I disabled this, because I can not support it at the moment. And I think this is not an advection method. It is using the same numerical method (a compositional field), just different terms in the equation (a different assembler).